include/EASTL/string_view.h

```diff
@@ -158,9 +158,19 @@ namespace eastl
 			return this_type(mpBegin + pos, count);
 		}
 
-		EA_CONSTEXPR int compare(basic_string_view sw) const EA_NOEXCEPT
+		static EA_CPP14_CONSTEXPR int compare(const T* pBegin1, const T* pEnd1, const T* pBegin2, const T* pEnd2)
 		{
-			return Compare(mpBegin, sw.data(), eastl::min_alt(size(), sw.size()));
+			const ptrdiff_t n1   = pEnd1 - pBegin1;
+			const ptrdiff_t n2   = pEnd2 - pBegin2;
+			const ptrdiff_t nMin = eastl::min_alt(n1, n2);
+			const int       cmp  = Compare(pBegin1, pBegin2, (size_t)nMin);
+
+			return (cmp != 0 ? cmp : (n1 < n2 ? -1 : (n1 > n2 ? 1 : 0)));
```

Comment:
The return value of compare should be taking care of this.  Why are these conditionals necessary?

Reply:
Because it *is* 'compare' of string_view. If yor meant 'Compare' instead than it's defined in char_traits.h and it's full implementation is
```
inline int Compare(const char8_t* p1, const char8_t* p2, size_t n)
	{
		return memcmp(p1, p2, n);
	}
```
which oviously works right only for strings of equal length. 
Also this code is what *precisely* eastl::string does as well, see
https://github.com/electronicarts/EASTL/blob/master/include/EASTL/string.h#L2970

Reply:
Ah gotcha.  Thanks.